Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnusedEnumeratedRule.swift

```diff
@@ -44,17 +56,87 @@ private extension UnusedEnumeratedRule {
                 return
             }
 
-            let position: AbsolutePosition
-            let reason: String
-            if firstTokenIsUnderscore {
-                position = firstElement.positionAfterSkippingLeadingTrivia
-                reason = "When the index is not used, `.enumerated()` can be removed"
+            addViolation(
+                zeroPosition: firstTokenIsUnderscore ? firstElement.positionAfterSkippingLeadingTrivia : nil,
+                onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
+            )
+        }
+
+        override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
+            guard node.isEnumerated,
+                  let trailingClosure = node.parent?.parent?.as(FunctionCallExprSyntax.self)?.trailingClosure,
+                  node.parent?.as(MemberAccessExprSyntax.self)?.declName.baseName.text != "filter"
```

Comment:
Add `let parent = node.parent` and check the last condition first.

Reply:
So I didn't go the `ShorthandArgumentRule` route. Instead, when I see a closure that is not the one I'm looking for, I  push it onto the stack anyway, and then stop counting `$0`'s and `$1`'s. Seems to work ok for that test case.